source/backend/api/Services/DocumentSyncService.cs

```diff
@@ -94,113 +106,122 @@ public ExternalBatchResult SyncMayanMetadataTypes(SyncModel model)
+                // Disable the document types that are not on the sync model
+                foreach (var documentType in pimsDocumentTypes)
                 {
-                    if (model.DocumentTypes.FirstOrDefault(x => x.Label == documentType.Label) == null)
+                    if (!model.DocumentTypes.Any(x => x.Name == documentType.DocumentType))
                     {
-                        deleteTasks.Add(mayanDocumentRepository.TryDeleteDocumentTypeAsync(documentType.Id));
+                        // TODO: currently there is no is_disabled field available for document types.
                     }
                 }
-
-                Task.WaitAll(deleteTasks.ToArray());
-                foreach (var task in deleteTasks)
-                {
-                    batchResult.DeletedDocumentType.Add(task.Result);
-                }
             }
+            documentTypeRepository.CommitTransaction();
 
-            SyncDocumentTypeMetadataTypes(model, ref batchResult);
-
-            return batchResult;
+            return new DocumentSyncResponse() { Added = createdDocumentTypes, Updated = updatedDocumentTypes, Deleted = deletedDocumentTypes };
         }
 
-        public async Task<DocumentTypeSyncResponse> SyncBackendDocumentTypes(SyncModel model)
+        public ExternalBatchResult SyncPimsToMayan(SyncModel model)
         {
-            this.Logger.LogInformation("Synchronizing Pims DB and Mayan document types");
+            this.Logger.LogInformation("Synchronizing Pims DB and Mayan document types (as well as attached metadata)");
             this.User.ThrowIfNotAuthorizedOrServiceAccount(Permissions.DocumentAdmin, this.keycloakOptions);
 
-            ExternalResult<QueryResult<DocumentType>> mayanResult = await mayanDocumentRepository.TryGetDocumentTypesAsync(pageSize: 5000);
+            ExternalBatchResult batchResult = new();
 
-            if (mayanResult.Status != ExternalResultStatus.Success && mayanResult.Payload?.Results?.Count == 0)
-            {
-                return new DocumentTypeSyncResponse();
-            }
+            Task<ExternalResult<QueryResult<DocumentType>>> documentTypeTask = mayanDocumentRepository.TryGetDocumentTypesAsync(pageSize: 5000);
+            documentTypeTask.Wait();
+
+            ExternalResult<QueryResult<DocumentType>> mayanDocumentTypes = documentTypeTask.Result;
 
             IList<PimsDocumentTyp> pimsDocumentTypes = documentTypeRepository.GetAll();
 
-            // Add the document types not in the backend
-            IList<PimsDocumentTyp> createdDocumentTypes = new List<PimsDocumentTyp>();
-            IList<PimsDocumentTyp> updatedDocumentTypes = new List<PimsDocumentTyp>();
-            foreach (var mayanDocumentType in mayanResult.Payload.Results)
+            // Add the document types not in mayan
+            IList<AddDocumentToMayanWithNameResponseModel> createResponseTasks = new List<AddDocumentToMayanWithNameResponseModel>();
```

Comment:
nit: createResponseTasks => createRequestTasks 
These tasks are to create new entries in mayan.